src/coord/rmr/rq.c

```diff
@@ -256,21 +90,30 @@ MRWorkQueue *RQ_New(int maxPending) {
   q->pendingInfo.head = NULL;
   q->pendingInfo.warnSize = 0;
   uv_mutex_init(&q->lock);
-  uv_async_init(uv_default_loop(), &q->async, rqAsyncCb);
-  q->async.data = q;
+  q->id = id;
   return q;
 }
 
+void RQ_Free(MRWorkQueue *q) {
+  uv_mutex_lock(&q->lock);
+  // clear the queue
+  queueItem *cur = q->head;
+  while (cur) {
+    queueItem *next = cur->next;
+    rm_free(cur);
+    cur = next;
+  }
```

Comment:
Can we get here with jobs in the queue, now that we have properly closed the UV thread?